### ISSUE TITLE
Explanation for use of two target lightings in one forward pass during training

### ISSUE THREAD

**Issue by RaphaelKreft** [Haian-Jin/Neural_Gaffer#20]
Dear NeuralGaffer Team, 

I recently looked at the project for a student project and I find it very interesting. Great work! 
When I studied the code of the training for image relighting  `neural_gaffer_training.py` there were two questions that arised for me and that I could not find an answer in the paper and I would be really thankful if you could enlight me:

1. In one training step one specific object is relighted. I expected to see that there is one target lighting condition so the network predicts the object under one target lighting. Actually, each training step involves the prediction of the same object in two target lighting conditions. I wonder why this is done. Did you also try with only one target lighting?

2. In the training step, when the hdr map, the ldr map and the condition image are prepared, they are encoded with the vae. I was surprised to see that  in the following code (line 701) the latents are gathered differently `.sample()` vs `.mode()`.  Why is this the case?

```
gt_latents = vae.encode(gt_image).latent_dist.sample().detach()
gt_latents = gt_latents * vae.config.scaling_factor # follow zero123, only target image latent is scaled

img_latents = vae.encode(input_image).latent_dist.mode().detach()   
target_envir_map_ldr_latents = vae.encode(target_envir_map_ldr).latent_dist.sample().detach()
target_envir_map_hdr_latents = vae.encode(target_envir_map_hdr).latent_dist.mode().detach()
```

Thank you very much already for your help.
Best regards

**Comment by Haian-Jin** [comment]
Hi, Thanks for your interests.

1. Both ways should work. I did as you described above to improve the data loading efficiency, so that the input images are loaded once, but but the loss are computed from two target lighting.
2. .sample() vs .mode() are something like training mode vs. infer mode. You can check its doc to see more.

**Comment by RaphaelKreft** [comment]
Hi,
many thanks for your response :)
The 1st point is now clear to me.
Regarding the use of `.sample()` vs `.mode()` I am getting what they are doing technically, but I am still wondering why you use **`.mode()` to get hdr latents** while you use **`.sample()`for ldr latents**

I would be really thankful to get an insight on that.
Best regards
